receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go

```diff
@@ -42,23 +42,11 @@ type PrometheusScraper struct {
 	host                     component.Host
 	clusterNameProvider      clusterNameProvider
 	simplePrometheusReceiver receiver.Metrics
+	leaderElection           *LeaderElection
+	running                  bool
 }
 
-func NewPrometheusScraper(ctx context.Context, telemetrySettings component.TelemetrySettings, nextConsumer consumer.Metrics, host component.Host, clusterNameProvider clusterNameProvider) (*PrometheusScraper, error) {
-	// TODO: need leader election
-
-	k8sClient := k8sclient.Get(telemetrySettings.Logger)
-	if k8sClient == nil {
-		return nil, errors.New("failed to start k8sapiserver because k8sclient is nil")
-	}
-
-	// get endpoint
-	endpoint := k8sClient.GetClientSet().CoreV1().RESTClient().Get().AbsPath("/").URL().Hostname()
-
-	// TODO: test RBAC permissions?
-
-	spFactory := simpleprometheusreceiver.NewFactory() // TODO: pass this in?
-
+func NewPrometheusScraper(ctx context.Context, telemetrySettings component.TelemetrySettings, endpoint string, nextConsumer consumer.Metrics, host component.Host, clusterNameProvider clusterNameProvider, leaderElection *LeaderElection) (*PrometheusScraper, error) {
```

Comment:
nit should we validate that the leaderElection is not nil?

Reply:
Good call, I am adding that behavior and a test in my follow up PR